### DISCUSSION TITLE
Provide documentation on authentication

### DISCUSSION THREAD

**Discussion by tomkerkhove** [scale-tone/DurableFunctionsMonitor#101] [Q&A] [answered]
We currently use Durable Functions Monitor in read-only mode but would like to integrate authentication.

However, it's not 100% how it works and how to configure it. What are the options?

**Comment by scale-tone** [comment, marked as answer]
Info on how to protect your standalone DfMon endpoint is [documented here](https://github.com/scale-tone/DurableFunctionsMonitor/tree/master/durablefunctionsmonitor.dotnetbackend#how-to-run). See the section on how to deploy it as a separate Function App instance. Additionally please see this [Config Setting Reference](https://github.com/scale-tone/DurableFunctionsMonitor/tree/master/durablefunctionsmonitor.dotnetbackend#config-setting-reference).

Programmatic config options that are available to your code in 'injected' mode are [described here](https://www.nuget.org/packages/DurableFunctionsMonitor.DotNetBackend). These options mostly map the app config settings, so normally you go either one way or another.

In short, in 'injected' mode DfMon endpoint *inherits* authentication settings of your app, whatever those settings are. You can then adjust DfMon's *authZ* behavior with DfMon-specific settings like `DFM_ALLOWED_USER_NAMES`, but there's no way for you to override main application's *authN* settings.

Does it explain that enough, or you still fell like something is missing in the above-mentioned docs?

**Reply by tomkerkhove** [reply]
Thank you for the information!

The info is a bit hidden in the folder but that is probably so that it is included on the NuGet package?

**Reply by scale-tone** [reply]
Yes, the intention was to keep all info specific to 'injected' mode close to the [NuGet package](https://www.nuget.org/profiles/durablefunctionsmonitor) (with all the proper references from the main GitHub page, of course).
Whatever is not specific to 'injected' mode - that is documented here, on GitHub.
